beacon-chain/light-client-network.md

```diff
@@ -0,0 +1,197 @@
+Participants only process messages that are within their chosen radius boundary.
+
+- Each gossip participant has a `node_id`. A `node_id` is a 256 bit unsigned integer, i.e. `0 <= node_id < 2**256`.
+- There is a distance function. It measures the distance between two `node_id`. It also measures the distance between `node_id` and `content_id`.
+
+#### Validation
+Validating `LightClientFinalityUpdate` and `LightClientOptimisticUpdate` follows the gossip domain(gossipsub) [consensus specs](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md#the-gossip-domain-gossipsub).
+
+### Data
+
+#### Types
+
+* LightClientBootstrap
+* LightClientUpdate
+* LightClientFinalityUpdate
+* LightClientOptimisticUpdate
+
+All data types are specified in light client [sync protocol](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#containers).
+
+#### Retrieval
+
+* Requests the `LightClientBootstrap` structure corresponding to a given post-Altair beacon block root.
+* `LightClientUpdatesByRange` - requests the `LightClientUpdate` instances in the sync committee period range [start_period, start_period + count), leading up to the current head sync committee period as selected by fork choice.
+* The latest `LightClientFinalityUpdate` known by a peer.
+* The latest `LightClientOptimisticUpdate` known by a peer.
+
+## Specification
+
+### Distance
+
+Nodes in the light client network are represented by their [EIP-778 Ethereum Node Record (ENR)](https://eips.ethereum.org/EIPS/eip-778) from the Discovery v5 network.
+A node's `node-id` is derived according to the node's identity scheme, which is specified in the node's ENR. A node's `node-id` represents its address in the DHT.
+
+The `node-id` is a 32-byte identifier. We define the `distance` function that maps a pair of `node-id` values to a 256-bit unsigned integer identically to the Discovery v5 network.
+
+```
+distance(n1, n2) = n1 XOR n2
+```
+
+Similarly, we define a `logdistance` function identically to the Discovery v5 network.
+
+```
+logdistance(n1, n2) = log2(distance(n1, n2))
+```
+
+### Content: Keys and Values
+
+The beacon chain light client DHT stores the following data items:
+
+* LightClientBootstrap
+* LightClientUpdate
+* LightClientFinalityUpdate
+* LightClientOptimisticUpdate
+
+Each of these data items is represented as a key-value pair. Denote the key for a data item by `content-key`. Denote the value for an item as `content`.
+
+All `content_key` values are encoded and decoded as an [`SSZ Union`](https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md#union) type.
+```
+content_key = Union[light_client_bootstrap_key, light_client_update_key, light_client_finality_update_key, light_client_optimistic_update_key]
+serialized_content_key = SSZ.serialize(content_key)
```

Comment:
You might take a looks at https://github.com/ethereum/portal-network-specs/pull/164 and update this to be inline with the changes I made there.  I removed the `Union` approach to content key creation in favor of explicitly defining the `content_key` for each data type.

My reasoning for this is that the current `Union` approach relies implicitly on the serialization semantics of SSZ Union types which doesn't feel correct to me.  I also think that the Union approach adds a level of indirection to how each `content_key` is structured that make it a little harder to initially ingest the specification.